compiler/rustc_codegen_llvm/src/attributes.rs

```diff
@@ -55,6 +55,22 @@ pub fn sanitize(cx: &CodegenCx<'ll, '_>, no_sanitize: SanitizerSet, llfn: &'ll V
     if enabled.contains(SanitizerSet::HWADDRESS) {
         llvm::Attribute::SanitizeHWAddress.apply_llfn(Function, llfn);
     }
+    if enabled.contains(SanitizerSet::MEMTAG) {
+        // Check to make sure the mte target feature is actually enabled.
+        let sess = cx.tcx.sess;
+        let features = llvm_util::llvm_global_features(sess).join(",");
+        let mte_feature_enabled = features.rfind("+mte");
+        let mte_feature_disabled = features.rfind("-mte");
+
+        if mte_feature_enabled.is_none()
+            || (mte_feature_disabled.is_some()
+                && mte_feature_disabled.unwrap() > mte_feature_enabled.unwrap())
```

Comment:
Nit: I believe you don't need to unwrap here at all, you can just compare the `Option`s. In fact, because `Some` compares higher than `None`, you can drop the `is_some` check too, making this:

```rust
mte_feature_enabled.is_none() || (mte_feature_disabled > mte_feature_enabled)
```

Reply:
Ah this is cleaner -- I went ahead and made this change, thanks!